pkg/fanal/image/image.go

```diff
@@ -9,102 +9,56 @@ import (
 	"golang.org/x/xerrors"
 
 	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	ftypes "github.com/aquasecurity/trivy/pkg/types"
 )
 
-type options struct {
-	dockerd    bool
-	podman     bool
-	containerd bool
-	remote     bool
-}
-
-type Option func(*options)
+type RuntimeFunc func(ctx context.Context, imageName string, ref name.Reference, option types.ImageOptions) (types.Image, func(), error)
 
-func DisableDockerd() Option {
-	return func(opts *options) {
-		opts.dockerd = false
-	}
+var runtimeFuncs = map[ftypes.Runtime]RuntimeFunc{
+	ftypes.ContainerdRuntime: tryContainerdDaemon,
+	ftypes.PodmanRuntime:     tryPodmanDaemon,
+	ftypes.DockerRuntime:     tryDockerDaemon,
+	ftypes.RemoteRuntime:     tryRemote,
 }
 
-func DisablePodman() Option {
-	return func(opts *options) {
-		opts.podman = false
-	}
-}
+func WithRuntimes(runtimes ftypes.Runtimes) ([]RuntimeFunc, error) {
```

Comment:
I think it can be done in NewContainerImage. Is there any benefit in exporting this function?

Reply:
I moved this to `NewContainerImage`, and made the function private. The function was also renamed